# Вы уже видели исключения в предыдущем коде. Они происходят, когда что-то идет не так,
# из-за некорректного кода или ввода. Когда происходит исключение, программа немедленно останавливается.

# Следующий код вызывает исключение ZeroDivisionError, пытаясь разделить 7 на 0:

num1 = 7
num2 = 0
print(num1/num2)

# Исключение - это событие, которое происходит во время выполнения программы и нарушает ее нормальный ход.

# Разные исключения вызываются по разным причинам.
# Обычные исключения:

# ImportError: импорт не удался;
# IndexError: список индексируется с числом, выходящим за пределы диапазона;
# NameError: используется неизвестная переменная;
# SyntaxError: код не может быть правильно проанализирован;
# TypeError: функция вызывается на значении неподходящего типа;
# ValueError: функция вызывается на значении правильного типа, но с неподходящим значением.

# Python имеет несколько других встроенных исключений, таких как ZeroDivisionError и OSError.
# Сторонние библиотеки также часто определяют свои собственные исключения.